nav2_mppi_controller/include/nav2_mppi_controller/tools/parameters_handler.hpp

```diff
@@ -224,24 +223,32 @@ void ParametersHandler::setParam(
 }
 
 template<typename T>
-void ParametersHandler::setDynamicParamCallback(T & setting, const std::string & name)
+void ParametersHandler::setDynamicParamCallback(
+  T & setting, const std::string & name, ParameterType param_type)
 {
   if (get_param_callbacks_.find(name) != get_param_callbacks_.end()) {
     return;
   }
 
-  auto callback = [this, &setting, name](const rclcpp::Parameter & param) {
+  auto dynamic_callback = [this, &setting, name](const rclcpp::Parameter & param) {
       setting = as<T>(param);
 
       if (verbose_) {
         RCLCPP_INFO(logger_, "Dynamic parameter changed: %s", std::to_string(param).c_str());
       }
     };
 
-  addDynamicParamCallback(name, callback);
+  auto static_callback = [this, &setting, name](const rclcpp::Parameter & param) {
```

Comment:
If a static callback is created, then if someone attempts to set a static parameter through the dynamic parameter interface, it would `return true` on the change, even though its static and rejected the update. I would think that it should fail to find the callback and thus `return false`, just as if the parameter doesn't exist since this parameter is not able to be adjusted dynamically

Reply:
@aosmw following up on this item

Reply:
@SteveMacenski 

I've taken this further in the latest commit prompted by your previous comment.  I have snipped the commit message here too.

Provide a mechanism to populate an rcl_interfaces::msg::SetParametersResult
with the reasons for unsuccessful parameter setting, so that it may be
propagated back to a set parameter client.

The mechanism provides feedback when attempting to set undefined
parameters, static parameters and could be used to validate dynamic
parameters and provide a reason for rejection.

NOTE: This changes public interface of ParametersHandler class.
s/setDynamicParamCallback/setParamCallback
s/addDynamicParamCallback/addParamCallback which takes
a callback function that is able to populate a
rcl_interfaces::msg::SetParametersResult.

In order to indicate an unsuccessful parameter change and the
reason, callback functions should append a "\n"
to the reason before appending to if it is not empty.